lte/gateway/python/magma/mobilityd/ip_allocator_dhcp.py

```diff
@@ -435,30 +439,22 @@ def _release_dhcp_ip(self, ip_desc: IPDesc) -> None:
         dhcp_desc = self.get_dhcp_desc_from_store(mac, vlan)
         logging.info("Releasing dhcp desc: %s", dhcp_desc)
         if dhcp_desc:
-            call_args = [
-                DHCP_HELPER_CLI,
-                "--mac", str(mac),
-                "--vlan", str(vlan),
-                "--interface", self._iface,
-                "--json",
-                "release",
-                "--ip", str(ip_desc.ip),
-                "--server-ip", str(dhcp_desc.server_ip),
-            ]
-            ret = subprocess.run(
-                call_args,
-                capture_output=True,
-                check=False,
+            if len(self.dhcp_helper_procs) == self.MAX_DHCP_PROCS:
```

Comment:
Is this thread safe? By a race condition, can we get in a state where len(self.dhcp_helper_procs) > self.MAX_DHCP_PROCS, i.e., the list will not be inspected any more? Might >= be more reasonable to mitigate this?

Reply:
If there are no _release_dhcp_ip calls for some time MAX_DHCP_PROCS=3 potentially stuck processes can be kept for a long time - should this be prevented?

Reply:
I made this a bit safer by changing it to `while len(self.dhcp_helper_procs) >= self.MAX_DHCP_PROCS`. That way we should safely remove any excess processes every time we want to release an IP. I believe the logic would be thread-safe regardless. This is the only place where we touch `dhcp_helper_procs`, , and there should never be more than one instance of `IPAllocatorDHCP`.